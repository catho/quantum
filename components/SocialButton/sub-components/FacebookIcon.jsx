import React from 'react';
import Proptypes from 'prop-types';
import styled from 'styled-components';

const StyledSvg = styled.svg`
  display: inline-block;
`;

const FacebookIcon = ({ size, title }) => {
  const SVGOriginalHeight = 402;
  const SVGOriginalWidth = 582;

  return (
    <StyledSvg
      role="img"
      aria-labelledby="facebook-title"
      xmlns="http://www.w3.org/2000/svg"
      width={size}
      height={size}
    >
      <title id="facebook-title">{title}</title>
      <g>
        <rect
          fill="none"
          id="canvas_background"
          height={SVGOriginalHeight}
          width={SVGOriginalWidth}
          y="-1"
          x="-1"
        />
      </g>
      <g>
        <path
          fill="#ffffff"
          id="svg_2"
          d="m13,10l3,0l0,3l-3,0l0,7l-3,0l0,-7l-3,0l0,-3l3,0l0,-1.255c0,-1.189 0.374,-2.691 1.118,-3.512c0.744,-0.823 1.673,-1.233 2.786,-1.233l2.096,0l0,3l-2.1,0c-0.498,0 -0.9,0.402 -0.9,0.899l0,2.101z"
        />
      </g>
    </StyledSvg>
  );
};

FacebookIcon.defaultProps = {
  size: '48',
};

FacebookIcon.propTypes = {
  size: Proptypes.string,
  title: Proptypes.string.isRequired,
};

export default FacebookIcon;
