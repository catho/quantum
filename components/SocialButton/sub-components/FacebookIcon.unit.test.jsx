import React from 'react';
import renderer from 'react-test-renderer';
import FacebookIcon from './FacebookIcon';

describe('FacebookIcon component', () => {
  it('Should match the snapshot', () => {
    expect(
      renderer.create(<FacebookIcon title="test" />).toJSON(),
    ).toMatchSnapshot();
  });

  it('when size prop is set to 120', () => {
    expect(
      renderer.create(<FacebookIcon title="test" size="120" />).toJSON(),
    ).toMatchSnapshot();
  });
});
