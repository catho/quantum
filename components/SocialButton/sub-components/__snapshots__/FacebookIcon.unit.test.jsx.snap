// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookIcon component Should match the snapshot 1`] = `
.c0 {
  display: inline-block;
}

<svg
  aria-labelledby="facebook-title"
  className="c0"
  height="48"
  role="img"
  width="48"
  xmlns="http://www.w3.org/2000/svg"
>
  <title
    id="facebook-title"
  >
    test
  </title>
  <g>
    <rect
      fill="none"
      height={402}
      id="canvas_background"
      width={582}
      x="-1"
      y="-1"
    />
  </g>
  <g>
    <path
      d="m13,10l3,0l0,3l-3,0l0,7l-3,0l0,-7l-3,0l0,-3l3,0l0,-1.255c0,-1.189 0.374,-2.691 1.118,-3.512c0.744,-0.823 1.673,-1.233 2.786,-1.233l2.096,0l0,3l-2.1,0c-0.498,0 -0.9,0.402 -0.9,0.899l0,2.101z"
      fill="#ffffff"
      id="svg_2"
    />
  </g>
</svg>
`;

exports[`FacebookIcon component when size prop is set to 120 1`] = `
.c0 {
  display: inline-block;
}

<svg
  aria-labelledby="facebook-title"
  className="c0"
  height="120"
  role="img"
  width="120"
  xmlns="http://www.w3.org/2000/svg"
>
  <title
    id="facebook-title"
  >
    test
  </title>
  <g>
    <rect
      fill="none"
      height={402}
      id="canvas_background"
      width={582}
      x="-1"
      y="-1"
    />
  </g>
  <g>
    <path
      d="m13,10l3,0l0,3l-3,0l0,7l-3,0l0,-7l-3,0l0,-3l3,0l0,-1.255c0,-1.189 0.374,-2.691 1.118,-3.512c0.744,-0.823 1.673,-1.233 2.786,-1.233l2.096,0l0,3l-2.1,0c-0.498,0 -0.9,0.402 -0.9,0.899l0,2.101z"
      fill="#ffffff"
      id="svg_2"
    />
  </g>
</svg>
`;
